tests/test_mig_shared_install.py

```diff
@@ -109,38 +110,45 @@ def exists_timedatectl(value):
 class MigSharedInstall__generate_confs(MigTestCase):
     """Unit test helper for the migrid code pointed to in class name"""
 
+    def before_each(self):
+        self.output_path = TEST_OUTPUT_DIR
+
     def test_creates_output_directory_and_adds_active_symlink(self):
         symlink_path = temppath('confs', self)
         cleanpath('confs-foobar', self)
 
-        generate_confs(destination=symlink_path, destination_suffix='-foobar')
+        generate_confs(self.output_path, destination_suffix='-foobar')
 
         path_kind = self.assertPathExists('confs-foobar')
         self.assertEqual(path_kind, "dir")
         path_kind = self.assertPathExists('confs')
         self.assertEqual(path_kind, "symlink")
 
     def test_creates_output_directory_and_repairs_active_symlink(self):
+        expected_generated_dir = cleanpath('confs-foobar', self)
         symlink_path = temppath('confs', self)
-        output_path = cleanpath('confs-foobar', self)
-        nowhere_path = temppath('confs-nowhere', self, skip_clean=True)
+        nowhere_path = temppath('confs-nowhere', self)
         # arrange pre-existing symlink pointing nowhere
         os.symlink(nowhere_path, symlink_path)
 
-        generate_confs(destination=symlink_path, destination_suffix='-foobar')
+        generate_confs(self.output_path, destination_suffix='-foobar')
 
-        self.assertEqual(os.readlink(symlink_path), output_path)
+        generated_dir = os.path.realpath(symlink_path)
+        self.assertEqual(generated_dir, expected_generated_dir)
 
     def test_creates_output_directory_containing_a_standard_local_configuration(self):
         fixture_dir = fixturepath("confs-stdlocal")
         expected_generated_dir = cleanpath('confs-stdlocal', self)
         symlink_path = temppath('confs', self)
 
         generate_confs(
-            destination=symlink_path,
+            self.output_path,
             destination_suffix='-stdlocal',
             user='testuser',
             group='testgroup',
+            mig_code='/home/mig/mig',
```

Comment:
I guess these three should be AUTO or a non-default path like `/home/testuser/X` depending on what we want to test. I'll leave that for a follow-up PR.